modules/cudf/src/data_frame.ts

```diff
@@ -1022,16 +1015,10 @@ export class DataFrame<T extends TypeMap = any> {
    * // }
    * ```
    */
-  sin(memoryResource?: MemoryResource): DataFrame<T> {
-    const mapper = (map: SeriesMap<T>, name: string) => {
-      return {
-        ...map,
-        [name]: _invokeIfNumericSeries(
-          this.get(name),
-          () => { return Series.new(this._accessor.get(name).sin(memoryResource)); })
-      };
-    };
-    return new DataFrame(this.names.reduce(mapper, {} as SeriesMap<T>));
+  sin<P extends keyof T>(memoryResource?: MemoryResource) {
+    return new DataFrame(this.names.reduce(
+             (map, name) => ({...map, [name]: (this.get(name) as any).sin(memoryResource)}),
```

Comment:
Should we throw a more descriptive error here if the types are wrong? Right now it'll throw something like `TypeError: this.get(...).sin is not a function`

Reply:
Spoke with Paul, I assumed that `typescript` would be a requirement to use `node-rapids`, but this isn't the case.

Regardless, this PR should provide better typing if people do decide to use typescript (i.e return `never` if the result won't occur), but also giving useful `throw` messages as well.